app/form_models/publishers/vacancy_form_sequence.rb

```diff
@@ -63,6 +63,8 @@ def next_incomplete_step_subjects?
   end
 
   def not_validatable_steps
-    %i[subjects review].freeze
+    %i[subjects review].tap do |steps|
+      steps << :documents if @vacancy.supporting_documents.any?
+    end
```

Comment:
By setting `documents` step as one of the non-validatable steps in the vacancy form sequence, when checking if all steps are valid, the now "invalid" documents form won't interfere with it.